integrations/GA4/utils.js

```diff
@@ -149,19 +207,22 @@ function getDestinationEventProperties(
 function getDestinationItemProperties(products, item) {
   const items = [];
   let obj = {};
-  if (type(products) !== "array") {
-    logger.debug("Event payload doesn't have products array");
-  } else {
-    // get the dest keys from itemParameters config
-    // append the already created item object keys (this is done to get the keys that are actually top level props in Rudder payload but GA expects them under items too)
-    products.forEach((p) => {
-      obj = {
-        ...getDestinationEventProperties(p, itemParametersConfigArray),
-        ...((item && type(item) === "array" && item[0]) || {}),
-      };
-      items.push(obj);
-    });
-  }
+  const contextOp = type(products) !== "array" ? "properties" : "product";
+  const finalProducts = type(products) !== "array" ? [products] : products;
+  // get the dest keys from itemParameters config
+  // append the already created item object keys (this is done to get the keys that are actually top level props in Rudder payload but GA expects them under items too)
+  finalProducts.forEach((p) => {
+    obj = {
+      ...getDestinationEventProperties(
+        p,
+        itemParametersConfigArray,
+        contextOp,
+        true
+      ),
+      ...((item && type(item) === "array" && item[0]) || {}),
```

Comment:
Both of cases we are getting an empty object. Isn't it?

Reply:
from line no 168, we are calling `createItemProperty`, which is creating the items array, if that is created, it won't be an empty object.

Reply:
What I mean to say is `(item && type(item) === "array" && item[0])` this expression returns a boolean value and the other expression with it in or is an empty object.

Reply:
addressed